### ISSUE TITLE
Building flatpak v1.19 and when copy password Ubuntu crashes

### ISSUE THREAD

**Issue by igor2x** [pwsafe/pwsafe#1308]
1. Download [pwsafe flatpak manifest file](https://github.com/flathub/org.pwsafe.pwsafe/blob/master/org.pwsafe.pwsafe.yml)
2. Change file lines 97 and 98 with:
```
        tag: 1.19.0
        commit: a274299be312c7829de28998f7544e580820a8f0
```
3. On Ubuntu 24.04 I have built flatpak on Wayland according to the[ instructions.](https://github.com/pwsafe/pwsafe/blob/master/README.LINUX.FLATPAK.md)
4. Flatpak is successfully build and Password Safe v1.19 is started up.
5. I have tested flatpak version and it looks like working fine until... Double click on entry and in program's status bar is displayed that password was copied.
6. Opened Text editor and paste does not work.
7. Back to Password Safe flatpak and right click and Copy and now wait a second and Ubuntu crashes (I assume gnome-shell crashed), Ubuntu login window appears and I need to login into Ubuntu again. I can repeat step 5 to step 7 and Ubuntu always crashes.
8. I opened Terminal and typed in command: `flatpak run org.pwsafe.pwsafe &> file.txt` and when Ubuntu crashes and I relogin into Ubuntu and the content of file.txt is:

> Gdk-Message: 18:57:20.391: Error reading events from display: Connection reset by peer

9. I have updated all of the libraries to the latest tags and commits and repeated flatpak build and tested pwsafe and it crashes the same way.

Can someone confirm this issue?

**Comment by nobugshere** [comment]
@igor2x ,
> I have updated all of the libraries to the latest tags and commits...

Just out of curiosity, did you happen to upgrade wxWidgets to 3.2.5? ( I think that would be commit 573e56d432cc2c4b19a27aa5cad3e6045bfc542b ). It looks like they fixed some Wayland related bugs, including a copy/paste issue.  See: https://www.wxwidgets.org/news/2024/05/wxwidgets-3.2.5-released

This might also be related to #1307.

**Comment by igor2x** [comment]
@nobugshere, yes I updated the wxWidgets to 3.2.5.

---- 

Now I have reverted back changes to my first test to step 2, so changing only those two lines.

Now I did several tests:
1. In manifest file comment out line 97 with # as first character in line.
2. Then in line 98 replace commit string for every test with bellow one:

| Commit Date | Commit                                   | Does it work? | Notes           |
| ----------- | ---------------------------------------- | ------------- | ----------------|
| 2024-06-21  | f34a0054fd15a1fafe55ec3b2b6df5c74c5fb9dd | crash         |                 |
| 2024-06-01  | 783629b93c383354c53ebebbaac376c62acac7df | crash         |                 |
| 2024-06-01  | 1a51878595e601a5d751d459f66ce82044502ae6 | crash         |                 |
| 2024-06-01  | 6c82f435e155958c8988412aa1633a87862e1cee | crash         |                 |
| 2024-06-01  | 4fcb177b7ff1383ba96e3e446868aff7d5a490eb | crash         | first crashing  |
| 2024-05-31  | 967d0c4bfb6cacb898168fc39600cae0bb8ab147 | works fine    | last known good |
| 2024-05-29  | bb981b1be1d5455e97831f61d27327cb00d1b9e3 | works fine    |                 |

I have looked to the source code of "first crashing" and I see changes related to clipboard there.

----

By the way crash can be reproduced in simpler way, just do double click on entry several times and Ubuntu crash appears.

**Comment by igor2x** [comment]
Now I have tested with [official v1.19 deb file for Ubuntu 24.04](
https://github.com/pwsafe/pwsafe/releases/download/1.19.0/passwordsafe-ubuntu24-1.19-amd64.deb)

I did:
1. Install pwsafe from deb file: `sudo dpkg -i passwordsafe-ubuntu24-1.19-amd64.deb`
2. Error is returned of missing dependencies, install them: `sudo apt -f install`
3. Repeat step 1.
4. Start a program: `pwsafe`
5. Double click on entry few times and Ubuntu crashes.

It looks to me this is not flatpak specific problem.

**Comment by hxu1** [comment]
It crashes the entire Gnome desktop environment? Is this happening only on Wayland or X11 as well?

**Comment by hxu1** [comment]
Seems fine on KDE Plasma, I'm setting up a VM to test on Gnome.

**Comment by igor2x** [comment]
@hxu1, I can only test on Wayland on Ubuntu 24.04. X-org does not work in my case, when I select (Ubuntu x.org) at login, type in password and then session gets black. This is known issue at my laptop and it behaves this way for years know (several Ubuntu versions).

**Comment by hxu1** [comment]
According to the other issue (https://github.com/pwsafe/pwsafe/issues/1307#issuecomment-2165228533), Gnome on X11 isn't affected. 

I tested KDE Plasma on Wayland, and I can reproduce the issue where copy isn't working. Although Plasma isn't crashing. I'm investigating the cause of this issue...

**Comment by hxu1** [comment]
@igor2x Can you please check if the build from #1310 is working for you?

**Comment by igor2x** [comment]
@hxu1, excellent no more crash, working fine.

Details:
1. I have downloaded [pwsafe flatpak manifest file](https://github.com/flathub/org.pwsafe.pwsafe/blob/master/org.pwsafe.pwsafe.yml).
2. I have opened file and changed:
3. Lines 96, 97 and 98 to:

        url: https://github.com/hxu1/pwsafe/
        #tag: 1.18.2fp4
        commit: 0ddc01c1899001e9283d78c06f1e9b370cc6a853

4. Created flatpak. It created successfully.
5. Checked in Help | About to see if I really built from @hxu1 repository and I did.

![image](https://github.com/pwsafe/pwsafe/assets/10895030/93d7bd1f-e958-4b82-a190-23d596078f77)

6.  Double clicked on entry. In status bar it wrote that password was copied.
7. Changed program to Text Editor and pasted the password. It pasted successfully. Great.
8. Back to Password Safe. Double clicked on entry few times, that used to crash Ubuntu and now no crash at all. It looks problem is fixed.

@hxu1, thank you very much for fixing this issue.

**Comment by igor2x** [comment]
Fixed with https://github.com/pwsafe/pwsafe/pull/1310 and fix released in Password Safe v1.19.1.
